Signal/src/network/GiphyDownloader.swift

```diff
@@ -46,15 +51,27 @@ class GiphyAssetSegment {
     }
 
     public func append(data: Data) {
+        guard state == .downloading else {
```

Comment:
Why does an AssetSegment have multiple datas? 

I expected just one, represented by it's range (whereas the GiphyAsset has multiple data's via it's multiple AssetSegments.)

Reply:
Sorry I got interrupted by an interview.  I've just pushed a commit that unifies this.